### PR TITLE
Add requested information to "Single-Node Clusters"

### PR DESCRIPTION
**Pull request by jillian-maroket**
Related Jira issue: SURE-8081

Change scope (v1.3 and v1.4):
- Disadvantages: Removed "No multi-replica support"
- Prerequisites: Removed StorageClass-related info
- Added section "Replica Count of the Default StorageClass"
- Added section "Multiple Replicas on a Node with Multiple Disks"
- Added section "Upgrades and Maintenance"

Change scope (v1.2):
- Prerequisites: Removed StorageClass-related info
- Added section "Replica Count of the Default StorageClass"
- Added section "Upgrades and Maintenance"

### CODE REVIEW

**Review by bk201 — APPROVED** [review]

File: docs/advanced/singlenodeclusters.md [hunk 1]
```diff
@@ -1,33 +1,60 @@
...
+The default StorageClass `harvester-longhorn` has a replica count value of **3** for high availability. If you use `harvester-longhorn` in your single-node cluster, Longhorn is unable to create the default number of replicas, and volumes are marked as *Degraded* on the [embedded Longhorn UI](../troubleshooting/harvester.md#access-embedded-rancher-and-longhorn-dashboards). 
+
+To avoid this issue, you can perform either of the following actions: 
+
+- Change the [replica count](../install/harvester-configuration.md#installharvesterstorage_classreplica_count) of `harvester-longhorn` to **1** using a [Harvester configuration](../install/harvester-configuration.md) file. 
+
+- [Create a new StorageClass](../advanced/storageclass.md#creating-a-storageclass) with the **Number of Replicas** parameter set to **1**. Once created, locate the new StorageClass in the list and then select **⋮** > **Set as Default**. 
+
+## Multiple Replicas on a Node with Multiple Disks 
+
```
Comment (bk201) [inline docs/advanced/singlenodeclusters.md:43]: @WebberHuang1118 can you double-check this section?

File: docs/advanced/singlenodeclusters.md [hunk 2]
```diff
@@ -1,33 +1,60 @@
...
+- Network: [Configure ports](../install/requirements.md#port-requirements-for-harvester-nodes) based on the type of traffic to be transmitted among VMs.
 
-  :::info important
-  The default StorageClass `harvester-longhorn` has a replica count value of `3` for high availability. If you use this StorageClass to create volumes for your single-node cluster, Longhorn is unable to create the configured number of replicas. This results in volumes being marked as "Degraded" on the Longhorn UI.
-  :::
+## Replica Count of the Default StorageClass 
+
+Harvester uses StorageClasses to describe how Longhorn must provision volumes. Each StorageClass has a parameter that defines the number of replicas to be created for each volume. 
+
+The default StorageClass `harvester-longhorn` has a replica count value of **3** for high availability. If you use `harvester-longhorn` in your single-node cluster, Longhorn is unable to create the default number of replicas, and volumes are marked as *Degraded* on the [embedded Longhorn UI](../troubleshooting/harvester.md#access-embedded-rancher-and-longhorn-dashboards).
```
Comment (albinsun) [inline docs/advanced/singlenodeclusters.md:34]: We can observe `Degraded` on the **Volumes** page directly. No need to enter LH.
image

Comment (jillian-maroket) [reply docs/advanced/singlenodeclusters.md]: Thank you. I have updated the doc.

**Review by albinsun — APPROVED** [review]
LGTM
But needs @WebberHuang1118 to review section _Multiple Replicas on a Node with Multiple Disks_.
Ref. https://github.com/harvester/docs/pull/617/files#r1703864152

**Review by WebberHuang1118 — APPROVED** [review]
LGTM, just a nit, thanks.

File: docs/advanced/singlenodeclusters.md [hunk 3]
```diff
@@ -1,33 +1,60 @@
...
+
+In high-availability clusters, **Replica Hard Anti-Affinity** ensures volume redundancy. However, this same mechanism can cause volumes to become degraded in single-node clusters (since no other nodes are available for scheduling of new replicas).
+
+If you want Longhorn to create multiple replicas on a node with multiple disks, perform the following actions: 
+
+1. Enable [`Replica Node Level Soft Anti-Affinity`](https://longhorn.io/docs/1.7.0/references/settings/#replica-node-level-soft-anti-affinity): When this setting is enabled, Longhorn schedules new replicas on nodes with existing healthy replicas of the same volume.
+
+1. Disable [`Replica Disk Level Soft Anti-Affinity`](https://longhorn.io/docs/1.7.0/references/settings/#replica-disk-level-soft-anti-affinity): When this setting is disabled, Longhorn does not schedule new replicas on disks with existing healthy replicas of the same volume. Disabling this setting provides failure tolerance for disks in single-node clusters.
+
+1. [Create a new StorageClass](../advanced/storageclass.md#creating-a-storageclass) and specify the disk tags that must be matched during volume scheduling.
```
[the remark this answers is not in the document]
Comment (jillian-maroket) [reply docs/advanced/singlenodeclusters.md]: Thank you. I have marked the step as optional.

**Pull request merged**